spyvm/test/test_primitives.py

```diff
@@ -492,6 +492,17 @@ def test_image_name():
     w_v = prim(primitives.IMAGE_NAME, [2])
     assert w_v.bytes == list(IMAGENAME)
 
+def test_set_image_name(monkeypatch):
+    def set_image_name(new_name):
+        assert new_name == "anImage.image" or new_name == "newImageName.image"
+    monkeypatch.setattr(space, "set_image_name", set_image_name)
+    w_r = prim(primitives.IMAGE_NAME, [2, "newImageName.image"])
+    assert space.unwrap_int(w_r) == 2
+    # revert
+    space.system_attributes[1] = IMAGENAME
+
+    monkeypatch.undo()
```

Comment:
It is not required to undo the monkeypatching done with the funcarg-monkeypatch at the end of a test. It happens automatically after the test function has returned.